crates/ra_syntax/src/ast/extensions.rs

```diff
@@ -65,6 +65,20 @@ impl ast::Attr {
             None
         }
     }
+
+    pub fn as_key_value(&self) -> Option<(SmolStr, SmolStr)> {
+        let tt = self.value()?;
+        let tt_node = tt.syntax();
+        let attr = tt_node.children_with_tokens().nth(1)?;
+        if attr.kind() == IDENT {
+            let key = attr.as_token()?.text().clone();
+            let val_node = tt_node.children_with_tokens().find(|t| t.kind() == STRING)?;
+            let val = val_node.as_token()?.text().trim_start_matches("\"").trim_end_matches("\"");
```

Comment:
hmm it seems to me we should have a method that returns the value of a string node properly already?

Reply:
mmm from my understanding `as_atom()` can be used with single-identifier attributes like `#[test]`, `as_call()` is useful for  function-like attributes and `as_named()` returns a single string... This new method can be used to parse key/value pairs. Trimming unfortunately seems necessary because `text()` includes the `"`

Reply:
What I meant was a function that would take the string token and properly return its value as a string, with `\` escapes and so on handled properly, but apparently we don't have that yet, at least I couldn't find it.